src/helpers/util.js

```diff
@@ -39,5 +39,40 @@ module.exports = {
       ret.push(row)
     }
     return ret
+  },
+
+  /*
+    Binary Search:
+    Assumes that @arg array is sorted increasingly
+    returns the smallest i such that target >= changes[i]
```

Comment:
We could reword it to `smallest i such that changes[i] < target does not hold, or changes.length if all elements are < target. It returns the index where target could be inserted to maintain the order of the array.` - then we do not need any special conditions.